### PR TITLE
hook task(s) now execute after inventory generation

### LINKED ISSUE #652: inventory file won't be generated before the hook task execute
v1.6.0b1 changed the way to generate inventory file in v1.5.5a16, which used to be a task "inventory_gen".
The result is my postup task won't get the inventory file and lead to failed to find any hosts to run it.

### PR DESCRIPTION
**Pull request by samvarankashyap**
Moved postup state to cli to make sure post provision hooks occur after the inventory generation

fixes #652

### CODE REVIEW

File: linchpin/cli/__init__.py [hunk 1]
```diff
@@ -401,6 +402,8 @@ def lp_up(self, targets=(), run_id=None, tx_id=None, inv_f="cfg"):
                     self._write_distilled_context(run_data)
         self._write_latest_run()
         self._write_to_inventory(inv_format=inv_f)
+        if ('post' in self.pb_hooks) and (self.__meta__ == "CLI"):
+            self.hook_state = '{0}{1}'.format('post', 'up')
```
Comment (p3ck) [inline linchpin/cli/__init__.py]: Are there down hooks as well?

Comment (herlo) [reply linchpin/cli/__init__.py]: @p3ck yes. There's 4 specific hooks: 'preup', 'postup', 'predestroy', 'postdestroy'. You can see an example of them here: https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/docs/source/examples/workspace/PinFile.complex.yml#L22

**Comment by herlo** [comment]
@samvarankashyap  I've updated the task to a more appropriate title. Please use more descriptive titles to help identify the issue/PR. Thanks!

**Comment by herlo** [comment]
@samvarankashyap It looks like this mostly works, except for postdestroy doesn't execute. I used this [topology](https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/docs/source/examples/workspace/PinFile.complex.yml#L22) if it helps. The pre tasks work, and the postup task succeeds, but the postdestroy task does not report.

```
$ linchpin -vvvp PinFile.complex.yml destroy libvirt-new
Processing target: libvirt-new
using data from run_id: None
rundb_id: 323
uhash: e771b0
calling: predestroy
hook predestroy initiated
running predestroy hooks
-------
start hook shell:hello
hello

end hook shell:hello
-------

.. snip ..

PLAY RECAP **********************************************************************************************
localhost                  : ok=25   changed=6    unreachable=0    failed=0   

Action 'destroy' on Target 'libvirt-new' is complete

ID: 1036
Action: destroy

Target              	Run ID	uHash	Exit Code 
-------------------------------------------------
libvirt-new         	   323	e771b0	        0

(pr653) [herlo@x99 (pr/653) workspace]$ 

```

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**